plugins/outputs/prometheus_client/prometheus_client.go

```diff
@@ -171,7 +174,12 @@ func (p *PrometheusClient) Write(metrics []telegraf.Metric) error {
 					"key: %s, labels: %v,\nerr: %s\n",
 					mname, l, err.Error())
 			}
-			p.metrics[desc.String()] = metric
+
+			metricWithTime := &MetricWithExpiration{
```

Comment:
don't declare a tmp var for metricWithTime, just do `p.metrics[desc.String()] = &MetricWithExpiration{ ....`